openmc/plots.py

```diff
@@ -0,0 +1,850 @@
+        cv.check_length('plot width', width, 2, 3)
+        self._width = width
+
+    @origin.setter
+    def origin(self, origin):
+        cv.check_type('plot origin', origin, Iterable, Real)
+        cv.check_length('plot origin', origin, 3)
+        self._origin = origin
+
+    @pixels.setter
+    def pixels(self, pixels):
+        cv.check_type('plot pixels', pixels, Iterable, Integral)
+        cv.check_length('plot pixels', pixels, 2, 3)
+        for dim in pixels:
+            cv.check_greater_than('plot pixels', dim, 0)
+        self._pixels = pixels
+
+    @filename.setter
+    def filename(self, filename):
+        cv.check_type('filename', filename, str)
+        self._filename = filename
+
+    @color_by.setter
+    def color_by(self, color_by):
+        cv.check_value('plot color_by', color_by, ['cell', 'material'])
+        self._color_by = color_by
+
+    @type.setter
+    def type(self, plottype):
+        cv.check_value('plot type', plottype, ['slice', 'voxel'])
+        self._type = plottype
+
+    @basis.setter
+    def basis(self, basis):
+        cv.check_value('plot basis', basis, _BASES)
+        self._basis = basis
+
+    @background.setter
+    def background(self, background):
+        self._check_color('plot background', background)
+        self._background = background
+
+    @colors.setter
+    def colors(self, colors):
+        cv.check_type('plot colors', colors, Mapping)
+        for key, value in colors.items():
+            cv.check_type('plot color key', key, (openmc.Cell, openmc.Material))
+            self._check_color('plot color value', value)
+        self._colors = colors
+
+    @mask_components.setter
+    def mask_components(self, mask_components):
+        cv.check_type('plot mask components', mask_components, Iterable,
+                      (openmc.Cell, openmc.Material))
+        self._mask_components = mask_components
+
+    @mask_background.setter
+    def mask_background(self, mask_background):
+        self._check_color('plot mask background', mask_background)
+        self._mask_background = mask_background
+
+    @show_overlaps.setter
+    def show_overlaps(self, show_overlaps):
+        cv.check_type('Show overlaps flag for Plot ID="{}"'.format(self.id),
+                      show_overlaps, bool)
+        self._show_overlaps = show_overlaps
+
+    @overlap_color.setter
+    def overlap_color(self, overlap_color):
+        self._check_color('plot overlap color', overlap_color)
+        self._overlap_color = overlap_color
+
+    @level.setter
+    def level(self, plot_level):
+        cv.check_type('plot level', plot_level, Integral)
+        cv.check_greater_than('plot level', plot_level, 0, equality=True)
+        self._level = plot_level
+
+    @meshlines.setter
+    def meshlines(self, meshlines):
+        cv.check_type('plot meshlines', meshlines, dict)
+        if 'type' not in meshlines:
+            msg = 'Unable to set on plot the meshlines "{0}" which ' \
```

Comment:
Question / Possible typo: Did you mean to say "set on plot the meshlines here"? Or did you mean "set the meshlines", "plot the meshlines"?